app/scripts/HiGlassComponent.js

```diff
@@ -2804,6 +2804,52 @@ class HiGlassComponent extends React.Component {
       track.removeViewportChanged = trackId =>
         this.removeScalesChangedListener(fromView, trackId);
       track.setDomainsCallback = (xDomain, yDomain) => {
+        if (!fromView) {
+          // If there is no `fromView`, then there must be an `initialXDomain` instead.
+          // Update the viewconfig to reflect the new `initialXDomain` array
+          // on the `viewport-projection-horizontal` track.
+          const { viewConfig } = this.state;
+          const newViewConfig = JSON.parse(JSON.stringify(viewConfig));
```

Comment:
Does this get called on every zoom event? If so, all this stringifying and parsing could have a significant effect on performance.

Reply:
Yes you are correct, this was unnecessary